src/v/cloud_storage/cache_service.cc

```diff
@@ -9,49 +9,110 @@
-  , _cnt(0) {}
+  , _max_cache_size(max_cache_size)
+  , _check_period(check_period)
+  , _cnt(0)
+  , _total_cleaned(0) {}
+
+uint64_t cache::get_total_cleaned() { return _total_cleaned; }
+
+ss::future<> cache::clean_up_cache() {
+    gate_guard guard{_gate};
+    co_await _walker.walk(_cache_dir.native());
+
+    if (_walker.get_current_cache_size() >= _max_cache_size) {
+        auto size_to_delete = _walker.get_current_cache_size()
+                              - (_max_cache_size * _cache_size_low_watermark);
+
+        auto candidates_for_deletion = _walker.get_files();
+
+        uint64_t deleted_size = 0;
+        size_t i_to_delete = 0;
+        while (deleted_size < size_to_delete) {
+            // todo(Lena): don't delete .part file since they may be in use
+            bool successfully_deleted = true;
+            auto file_to_remove = candidates_for_deletion[i_to_delete].filename;
+            vassert(
+              0
+                == file_to_remove.substr(0, _cache_dir.native().length())
+                     .compare(_cache_dir.native()),
+              "Tried to clean up {}, which is outside of cache_dir {}.",
+              file_to_remove,
+              _cache_dir.native());
+
+            try {
+                co_await ss::remove_file(file_to_remove);
+            } catch (std::exception& e) {
+                successfully_deleted = false;
+                vlog(
+                  cst_log.error,
+                  "Cache eviction couldn't delete {}: {}.",
+                  file_to_remove,
+                  e.what());
+            }
+            if (successfully_deleted) {
+                deleted_size += candidates_for_deletion[i_to_delete].size;
+            }
+            i_to_delete++;
+        }
+        _total_cleaned += deleted_size;
+        vlog(
+          cst_log.debug,
+          "Cache eviction deleted {} files of total size {}.",
+          i_to_delete,
+          deleted_size);
+    }
+}
 
 ss::future<> cache::start() {
+    // todo(Lena): delete leftover .part files
     vlog(cst_log.debug, "Starting archival cache service");
-    // todo: start maintenance cycle here
+
+    // TODO: implement more optimal cache eviction
+    _timer.set_callback([this] {
+        if (ss::this_shard_id() == 0) {
+            return clean_up_cache();
+        }
+        return ss::make_ready_future<>();
+    });
+
+    _timer.arm_periodic(_check_period);
     return ss::make_ready_future<>();
 }
 
 ss::future<> cache::stop() {
     vlog(cst_log.debug, "Stopping archival cache service");
-    return _gate.close();
+    _timer.cancel();
+    return _walker.stop().then([this] { return _gate.close(); });
 }
 
 ss::future<std::optional<cache_item>> cache::get(std::filesystem::path key) {
     gate_guard guard{_gate};
     vlog(cst_log.debug, "Trying to get {} from archival cache.", key.native());
     ss::file cache_file;
     try {
+        // TODO: update access time, this will be used by cache eviction
```

Comment:
nit: Add not about the access time tracking and that it's not yet targeted by the first cache implementation

Reply:
You suggest adding the comment in here or in some other place?